src/stream/stream/count.rs

```diff
@@ -0,0 +1,41 @@
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct CountFuture<S> {
+    stream: S,
+    count: usize,
+}
+
+impl<S> CountFuture<S> {
+    pin_utils::unsafe_pinned!(stream: S);
+    pin_utils::unsafe_unpinned!(count: usize);
```

Comment:
Can we rebase this branch and use `pin_project!` here?

Reply:
Yes, can do that. Sorry for the late reply, I was quite busy the last weeks.